math/mathcore/src/Fitter.cxx

```diff
@@ -847,14 +847,16 @@ bool Fitter::DoMinimization(std::unique_ptr<ObjFunc_t>  objFunc, const ROOT::Mat
 template<class ObjFunc_t>
 bool Fitter::DoWeightMinimization(std::unique_ptr<ObjFunc_t> objFunc, const ROOT::Math::IMultiGenFunction * chi2func) {
    // perform the minimization initializing the minimizer starting from a given obj function
-   // and apply afterwards the correction for weights. This applyies only for logL fitting
+   // and apply afterwards the correction for weights. This applies only for logL fitting
    this->fFitType = objFunc->Type();
    fExtObjFunction = nullptr;
-   fObjFunction = std::move(objFunc);
+   // need to use a temporary shared pointer to the objective function since we cannot use the unique pointer when it has been moved
+   std::shared_ptr<ObjFunc_t> sObjFunc{ std::move(objFunc)};
+   fObjFunction = sObjFunc;
    if (!DoInitMinimizer()) return false;
    if (!DoMinimization(chi2func)) return false;
-   objFunc->UseSumOfWeightSquare();
-   return ApplyWeightCorrection(*objFunc);
+   sObjFunc->UseSumOfWeightSquare();
+   return ApplyWeightCorrection(*sObjFunc);
```

Comment:
```suggestion
   fObjFunction->UseSumOfWeightSquare();
   return ApplyWeightCorrection(*fObjFunction);
```
When then problem is that `objFunc` is invalid because if was moved to `fObjFunc`, I think the simplest solution is to simply use that one. Then you don't need to create this extra shared pointer.

Reply:
The problem is the type required is of the derived class, you would need to do a dynamic_cast. I think is more efficient to just copy the pointer